### DISCUSSION TITLE
MAIANA-2

### DISCUSSION THREAD

**Discussion by peterantypas** [peterantypas/maiana#121] [Show and tell]
The work has begun. General layout of a fully self-contained wireless MAIANA here:

![Screen Shot 2023-01-07 at 9 12 01 AM](https://user-images.githubusercontent.com/1565933/211162561-b4f2a9d3-f827-476a-bdb5-6be05cfb5dc1.png)

It maintains the same transponder circuit and adds the ESP32-PICO-M2 module for WiFi. 

The battery is a single 18650, partially fit through a cutout in the PCB and accessed via SMD spring fingers (plus tape or heat shrink to hold it down). I expect about 12-16 hours of operation depending on whether it's an access point or a station. This should be enough for regular charter boat use.

Transmit power is delivered from a 330MF supercapacitor. It's the big blue cube in front of the battery.

The bottom fits a standard 1"-14 antenna mount. My original attempt to make this adapter for MAIANA was not very successful because of the weight center of the big unit. This one is fine. The USB-C charging connector will be exposed in the bottom:

![Screen Shot 2023-01-07 at 9 12 19 AM](https://user-images.githubusercontent.com/1565933/211162622-1e2f7ad7-8b8d-404f-b98c-8f78a1f22414.png)

**Comment by peterantypas** [comment]
If you're wondering about the button box on the exterior, this part is 3D printed out of ASA and contains a rubber seal for a pushbutton (from a standard flashlight design) as well as a red/green LED behind a press fitted light pipe. Its PCB connects to the interior via a JST cable, and the whole part is glued to the PVC exterior during assembly, sealing it completely. This is where 3M 5200 comes in handy.

**Comment by peterantypas** [comment]
First prototype of the button box. It needs some adjustments, but it fits well and matches the curve perfectly. Cosmetics were not a concern yet :)

![IMG_1777](https://user-images.githubusercontent.com/1565933/211886561-e4ac9163-9c79-4b5d-8e01-bd37d5e4de32.JPG)

**Comment by peterantypas** [comment]
Antenna tuning is done. Only needed a 160nH shunt inductor.

![Screen Shot 2023-01-18 at 5 48 00 PM](https://user-images.githubusercontent.com/1565933/213336949-591c568a-5c93-40d9-8868-16a917a9ba9e.png)

![IMG_1780](https://user-images.githubusercontent.com/1565933/213337013-02cbebbf-0b37-4219-81ef-3f9af035df47.jpg)

**Comment by peterantypas** [comment]
Today I did an A/B test for radiated power at 3 meters. There was 2 dB of difference between the big half-wave and the little quarter-wave antenna. It's definitely material, but I was expecting way more than 3 dB so I'll take it!

**Comment by peterantypas** [comment]
Since the materials are somewhat different, there was also a need to retune the GPS antenna. Thankfully, it was just a tiny change in a cap value.

![IMG_1788](https://user-images.githubusercontent.com/1565933/213885736-cbd4f992-a347-4f6d-bfde-c55347f752b5.JPG)
![Screen Shot 2023-01-21 at 12 14 01 PM](https://user-images.githubusercontent.com/1565933/213885750-7ea0a267-56ab-42c9-90be-1c55428a2984.png)

And this concludes all the tuning activities.

**Comment by peterantypas** [comment]
So after keeping them in my parts drawer for more than 2 years, I finally took the AFT05MS004N MOSFETs out for a test spin, as MAIANA will have to switch to this transistor now since the original is obsolete and no longer in distribution. These NXP transistors are designed for handheld VHF radios running from 7.4V lithium batteries so they're specified with a drain voltage of 6.5V minimum.

It took me about 2 hours to tune the input and output impedance, but no matter what I did, I could not get anything less than 3-3.5 watts out of it. This is thing is a touchy firecracker. So, after much frustration, I dropped the drain voltage below 7V and it moved in the right direction. And then I went below 6V. And I kept going until the power output settled at 2 watts with the drain at 4.5V. I checked the spectrum to 6 GHz to make sure I wasn't doing anything weird like transmitting parasitics. Nothing. It was just the normal harmonics with the 2nd and 3d at the expected levels.

So what does this mean? Well, on the surface it means that the new MAIANA can run entirely from a 5V source. And although it draws about 600mA during transmission, a $4 supercapacitor is more than capable of supplying the required current without taxing that 5V source too much. 

So does this mean a USB-powered MAIANA is possible? Probably. Is there a need for one?

**Reply by kagouraki** [reply]
I think that for the non portable maiana there is no need to be 5volt powered. You wont gain anything and you will loose usable cable lenght to the device. 
But the portable one could omit the battery and charge circuit and step up for less production cost and you can power directly from a phone power bank. Everybody has a power bank. Also there are some beefy ones who can run maiana way longer than just a 18650.

**Reply by peterantypas** [reply]
Interesting. How would you secure it? Strap the power bank somewhere in the cockpit with a cable running to the unit?

**Reply by muebau** [reply]
I think USB-powered is a great thing.
As most of the devices in the EU will become USB-powered (USB-C) I think we will see more and more equipment will tend to this way of power supply.
The problem of long cables and 5 volt and "high" current is real on the other hand.

To utilize USB PD (up to 20 volts) and step down to 5 volt seems senseless as it would work with 12 volts (current version) too.

The USB option with all the nice options like power banks, of-the-shelf solar chargers, etc. is a nice feature anyway.

**Reply by kagouraki** [reply]
Yes Peter. I think of it as a powerbank with a 2-3 meter good quality usb C cable somewhere in the cockpit. 
The portable version will be somehow strapped somewhere near the helm and cockpit because...well its portable and you install it, use it and then remove it. So it is reasonably to install it on arms lenght. Powerbank can live in a cupholder or a netpouch dudeki nearly everyone has. Let the people unwrap their creativity.
I dont think there is enough current draw to have enough voltagedrop to be a problem. You will have the supercap and you can adjust in your circuit how fast it will charge. You have plenty of time between the fastest trasmisions when moving.

**Reply by peterantypas** [reply]
Makes sense. The portable version fits on a 1-14" standard VHF antenna threaded mount. The idea is you screw it on, use it, unscrew it and plug in the USB cable for charging. But depending on the mount, you could fit a USB cable right through it and leave it permanently plugged in. My only concern is mechanical stability of the USB-C connector itself.

But you could also have the best of both worlds: Use the battery with a charger permanently plugged in, keeping it topped off.

**Reply by carlos-ddd** [reply]
Which kind of output matching capacitors did you choose for the AFT05MS004N when trying to get "the power" out of it? And are you sure about your (grounded) CPW dimensions (better asking which PCB-stackup are you currently aiming at?). Have you tried sourcing the TA0395A for RX performance? 

I'd love to see a portable version for charter usage with ESP32 WiFi so it's usable with Navionics on the iPad like "out of the box"!

**Comment by peterantypas** [comment]
First mostly-working prototype is running on the bench:

![IMG_1810](https://user-images.githubusercontent.com/1565933/221237324-ae9577b5-5849-4611-888e-f7068e90d424.JPG)

This is the most power-hungry configuration: Running as an Access Point serving a NMEA TCP client. Will still yield more than 12 hours of operation on a 3000mAh 18650.

**Comment by peterantypas** [comment]
First fully functioning prototype. Started at 7:17am with a fully charged battery, let's see if it makes it to 7 pm.

![IMG_1824](https://user-images.githubusercontent.com/1565933/224357511-82f6f3a9-fc62-43f0-a39a-eebe342a9b07.JPG)

**Reply by peterantypas** [reply]
So far so good after 10 hours. RX performance seems great, so I'm happy about that. Check out that target outside the Golden Gate. It's about 17NM from here:

<img width="1182" alt="Screen Shot 2023-03-10 at 5 23 27 PM" src="https://user-images.githubusercontent.com/1565933/224457478-655e1f74-4187-48ff-9c6a-47787e53091b.png">

**Reply by peterantypas** [reply]
On the new board, I added a custom GreenPAK IC which does many things, one of which is UVLO (under voltage lockout). The UVLO will cut off when the battery drops to 3.47V and will refuse to power the system back up until the voltage recovers. I don't know how long this will take, but I'll try to keep an eye on the unit until it does (hopefully before I go to bed). Although I added an external ADC for battery monitoring, I didn't do anything with it in the firmware yet, and I don't want to do that now so as not to disturb the experiment.

**Reply by peterantypas** [reply]
It took exactly 12.5 hours to shut down. That's after running it in AP mode with an occasional client consuming data. I'm going to charge it overnight and tomorrow configure it as a WiFi node to join a network instead and see how far it goes.

When plugging the cable in the bottom, the charging status LED is bright enough to see:

![IMG_1825](https://user-images.githubusercontent.com/1565933/224463981-bb2d220a-f2ca-4a84-b5ab-9d6f0c8d3ccd.JPG)

**Comment by peterantypas** [comment]
Conclusions from yesterday:

1. The MCP73830 is not a good charger IC for this device. It started limiting charge current way too early in the curve and ended up cutting off at 4.12 volts. Bottomline: It didn't really charge the battery in full and took way too long as well. Will find a different solution.
2. Under certain circumstances, the power-up sequence can result into the transponder circuit getting stuck. I will need to add a workaround for this in the HW design.

**Comment by peterantypas** [comment]
Alright, the verdict is that one more iteration is needed:

1. Instead of a battery charger, I will use a PMIC that can power the system while charging the battery.
2. There needs to be a reverse polarity protection scheme for the battery charger.
3. The supercap circuit needs to prevent discharge in either direction when the system is off. 
4. For on/off control, I will use a dedicated load switch from Toshiba that I designed into a wearable years ago. Typical P-Channel MOSFETs don't achieve low "on" resistance at low voltage and sub-1A current levels. I get a 40-60mV drop across the load switch. It should be under 10.

That's it. Everything else is firmware related.

**Reply by peterantypas** [reply]
I finally found the right 3D printing house for the adapter. They are the only one who has consistently produced the 1"-14 thread with Nylon 12 using SLS. So that part at least is finally done.

The plan is to use a two-part epoxy between the PVC tube and the adapter as well as the PVC tube and the button box (also Nylon 12). Assembly and disassembly will be done from the top by removing the TPU (rubber) cap.

**Reply by pmendiuk** [reply]
Will you still retain wired N2K capability with this design?

**Reply by peterantypas** [reply]
What would be the use case?

**Reply by pmendiuk** [reply]
I'm thinking about having the option to connect and power the transceiver via N2K or USB-C.  I regularly skipper a couple of boats that do not have WIFI or AIS and it would be nice to have the option of mounting the transceiver once for the entire trip.

**Comment by Layvack** [comment]
I don't know much about this subjet, but a Starlink dish have power supllied through a 75 feet cable, I think your projet could do the same and possibly use the Starlink cable, it has weather seals, you could vary the voltage to your needs, that would probably work for a permanent mount on a boat.

I don't know about the legallity, but doesn't AIS use VHF signal? How can it be illegal to send a VHS signal if you're certified to use a VHS radio, as long as you are not interfering with others?

**Comment by Layvack** [comment]
Starlink uses a Cat 6 networking cable and the dish/modem is supllied with PoE (power over ethernet) it needs 48volts. And like I said I don't know much about this, just thought I would mention this as a possible solution.

I asked OpenAI your first question and to resume what it said is this: "In summary, to choose output matching capacitors for the AFT05MS004N, you will need to calculate the required capacitance values based on the desired operating frequency and load impedance, and select capacitors that are suitable for high-frequency operation with low ESR and ESL values."

And toe the question: Is it possible to see a portable version for charter usage with ESP32 WiFi so it's usable with Navionics on the iPad or rpi4 like "out of the box"!

I received the following answer:


Yes, it is possible to create a portable chartplotter using an ESP32 with WiFi connectivity. Here's one approach you could consider:

  1.  Get an ESP32 development board with built-in WiFi, such as the ESP32 DevKitC.

  2.  Connect a GPS module to the ESP32 to obtain location data.

  3.  Write software to read GPS data, fetch map data from a map provider (such as OpenStreetMap), and display the map on an OLED display or TFT display.

  4.  Implement a communication protocol between the ESP32 and the iPad or RPi4, so that the iPad or RPi4 can send commands to the ESP32 and receive data, such as the current GPS location or user inputs.

  5.  If you want to use Navionics specifically, you'll need to reverse-engineer the Navionics protocol to communicate with their servers and fetch map data. This could be a challenging task, as Navionics likely uses proprietary protocols.

Alternatively, you could use a third-party library that can parse and display Navionics charts, such as the nv-chart library (https://github.com/nv-co/nv-chart). However, this library may not work "out of the box" with an ESP32, and some adaptation may be required.

Overall, creating a portable chartplotter using an ESP32 would require some programming skills and effort, but it is certainly feasible.


You can get access to OpenAI with ChatGPT membership is free, and OpenAI can even code for you. You can ask just about anything. I hope this helps, let me know how you are doing.

Regards

JC

**Comment by peterantypas** [comment]
My opinion of OpenAI’s understanding of RF design: Middle school student who memorized Wikipedia and thinks they know it all.

**Reply by carlos-ddd** [reply]
Not sure if your answer is meant kind of kinky or purely funny @Layvack ? I'm sorry if I was unfriendly or my comment was not welcome ...

- I was looking for information on the used output capacitors like the series (e.g. ATC100-B) since that is like the most important thing in RF output matching when growing into the 35 dBm output power region. Not well suited output capacitors will make your life a hell and they are like one of the key selection factors and show-stoppers in PA (power amplifier) development.
- And I was interested in the stackup since I saw the 0.1/1mm (w/h) grounded coplanar waveguides in the Eagle design which did not match my memories with 50R impedance on OSH-park's 4 layer stackup.
- And then I remembered having used the TA0395A filter from TaiSAW which is a nice 5x7mm SAW filter specially designed for AIS frequencies which would allow for a massive increase in RX large signal performance of any modem-chip like the Si44xx. That could (in my eyes) massivly increase receiver blocking and thus detection-range in situations where there are many (out of band) transmitters present. Unfortunately SAW filters are kind of hard to get these days, despite the datasheet existing. They are often produced based on customer demand (and thus are quite expensive). So having a source is key - and I'd have good hope sourcing them from municom. @peterantypas 

For the WiFi Access Point & Navionics part I wanted **to express my joy of seeing** that MAIANA-2 obviously is evolving excatly in this direction ("I like a lot!"). Navionics already supports taking AIS data from a generic device via TCP or UDP and @muebau has even gone through the work of creating a website GUI for the settings of MAINA doint excatly that (https://github.com/muebau/MAIANA-WiFi). Not sure why ChatGPT would make me start developing chartplotter SW ;)

I'd like to help if I can ...

**Comment by Layvack** [comment]
Don't worry Carlos, I just thought that it could help, I don't know anything about your project and all mentioned in your Email is over my head, but love what you are doing.

ChatGTP is just a program, it doesn't care what you do. I ask it what would be a cheap way to do something and it gave me parts numbers and how to make them work. I think it's worth a try. AI learns as it goes, and sometimes the professor gets bested by a student.

Regards
JC



________________________________

**Comment by peterantypas** [comment]
Quick update:

- I migrated the design to the TP4056 IC for battery charging. It seems to work better with 18650 batteries charged at 1A. 
- I think I finally resolved an issue with the 3D printed thread adapters coming back with missing sections. Seems to be a STEP vs STL problem so now only STL (mesh) files will be used for fabrication.

Third prototype is in fab.

**Comment by peterantypas** [comment]
I have decided to shelve this project for now.

The board's form factor creates a lot of problems with assembly and testing and I don't think it warrants the time and energy I will need to put into it.

**Reply by peterantypas** [reply]
That said, there were a few lessons learned and some of those lessons will likely apply to the next version of MAIANA:

- It's possible to make a decent antenna mount via 3D printing. It will likely require a redesign of the main board, antenna and case so they all fit together precisely.
- The new MOSFET and relatively inexpensive supercap will overcome pretty much any voltage drop across any length of Cat5.
- It's possible to get OK performance with a short 1/4 wave antenna, but only when the geometry of the ground plane is known. Any cable from the PCB to the cabin alters this geometry in unpredictable ways, so the existing 1/2 wave antenna remains the best path forward.

**Reply by muebau** [reply]
After I read #131 I wondered why not combine the ideas. What if you just use a PCB with an ESP32 and a 18650 battery (or batteries). It would be attached similar to the standard VHF mount adapter but just with a RJ45 plug.

Another could be to have a battery module and a WiFi module but It might become to long this way.

Just a spontaneous idea.

**Comment by Layvack** [comment]
ESP32 sounds like a good idea, it has WiFi and Bluetooth works on 3.3V DC 250 mA. With a small solar panel 52mmx52mmx3mm 3.3 volts 0.3 watts might be enough for the ESP32, for the rest I don't know.

It's good to see you're keeping it going,

Regards
JC
________________________________
